Frontend/src/components/mentor/mentor-profile/MentorProfile.jsx

```diff
@@ -208,7 +208,38 @@ function MentorProfile() {
               </Typography>
             </div>
           </div>
-          {isModalOpen && (
+          <Modal open={isModalOpen} onClose={handleModalToggle}>
+            <Box
+              sx={{
+                position: "absolute",
+                top: "50%",
+                left: "50%",
+                transform: "translate(-50%, -50%)",
+                width: "50%",
+                height: "75%",
+                bgcolor: "background.paper",
+                border: "1px solid #000",
+                borderRadius: "8px",
+                boxShadow: 24,
+                p: 4,
+                overflow: "auto",
+              }}
+            >
+              <MentorProfileModal
+                mentorData={userData}
+                handleDropdownToggle={handleDropdownToggle}
+                dropdownOpen={dropdownOpen}
+                selectedSkills={selectedSkills}
+                handleCheckboxChange={handleCheckboxChange}
+                skillsList={skillsList}
+                closeModal={() => {
+                  handleModalToggle();
+                  fetchMentorData();
+                }}
+              />
+            </Box>
+          </Modal>
+          {/* {isModalOpen && (
```

Comment:
remove commented code (or add a todo to remove it, if theres a really good reason to keep it around for now)